Bug  basket moving beyond play area

timotiousPrime opened the pull request:
Changed left offset to use percentage instead of px and fixed bug of basket leaving playArea

timotiousPrime committed:
Fixed BUG: basket moves beyond play area with keys

timotiousPrime committed:
Removed my majestic code that fixed the bugs

timotiousPrime committed:
created upper and lower bounds for basket.xPos

timotiousPrime committed:
Put bounds on the mutators for basket.xPos

timotiousPrime committed:
removed unnessasary console.log

timotiousPrime committed:
change xPos to % and fixed basket leaving playArea

tranhuuphucuong commented on rendering.js:
```diff
@@ -41,7 +42,29 @@ function renderOverlay({ gameMode }) {
+    // }
+
+    // Gets the of the basket in % value
+    let leftOffSetPercent = basket.xPos - halfOfBasket
+    if (leftOffSetPercent < 0 ) {
+        leftOffSetPercent = 0
+    } else if (leftOffSetPercent >= 100 - halfOfBasket*2) {
+        leftOffSetPercent = 100 - halfOfBasket*2
+    }
```
Great

poltak reviewed (approved):
LGTM! I wanna try and fix up the regressed collision algo on this branch (I think due to the unit change from px to %), then I'll merge it

poltak committed:
Add ID to play area div

- update all references + add constant

poltak committed:
Give left right movement mutations consistent bounds

- now they both move (100/8)% of the play area's width - i.e., press left 8 times and you move one screen
- removed param passing in new position

poltak committed:
Ensure collision detect fn gets px values, not percentages
